writing-your-first-es2015-javascript-module.md

```diff
@@ -0,0 +1,140 @@
+This configuration tells Babel that we want to use the [babel-preset-es2015](http://babeljs.io/docs/plugins/preset-es2015/) module that we installed.
+
+Having installed the Babel dependencies, let's next look at the directory structure of our `yarsay` module:
+
+## Structuring Your Project
+
+For my [ES2015 sample project](https://github.com/bcoe/yarsay) I used the following directory structure:
+
+* `src/`: contains the pre-transpiled ES2015 source files.
+* `lib/`: this directory contains the transpiled ES5 code (you probably shouldn't
+  add this directory to source control).
+* `test/`: the test directory (the tests exercise the ES2015 code in the
+  `/src` directory rather than the transpiled code).
+
+## Unit Tests and Coverage
+
+Babel provides the module `babel-register` which hooks into Node's require statement. This hook compiles your ES2015 code on the fly as your project needs it. `babel-register` is great for writing unit-tests, allowing you to exercise your ES2015 code directly without executing your build step.
+
+The test coverage tool [nyc](https://github.com/bcoe/nyc) integrates with `babel-register`, allowing you to add coverage reporting and transpilation in one fell swoop. Here's how this can be achieved:
+
+1. install your favorite unit testing framework (in my case `npm i mocha --save-dev`).
+2. install `nyc` (`npm i nyc --save-dev`), `nyc` will be used both for instrumenting
+  test coverage, and for compiling ES2015 code on the fly.
+3. add a `test` script to your `package.json` that looks something like this:
+
+```sh
+nyc --require=babel-core/register mocha
+```
+
+The flag `--require babel-core/register` tells nyc to include `babel-register` when executing the
+test suite, eliminating the need for a transpilation step when running your tests.
+
+## Packaging Your ES2015 Module
+
+It's nice to be able to run your unit-tests against the raw ES2015 code, but when publishing
+a module you want it to work on a variety of platforms, we achieve this by adding a `prepublish`
+build step to our ES2015 module. I'm a big fan of using npm scripts for this sort of thing.
+
+Here are the important parts of `yarsay`'s _package.json_:
+
+```json
+{
+  "scripts": {
+    "build": "cd src; babel ./*.js -d ../lib",
+    "prepublish": "npm run build",
+    "pretest": "standard ./src/*.js",
+    "test": "nyc --require babel-core/register mocha"
+  },
+  "main": "lib/yarsay.js",
+  "bin": "lib/cli.js"
+}
+```
+
+* `build`: enters the ES2015 source directory and executes the transpiler, and stores the final source-code in `/lib`.
+* `prepublish`: executes when you publish a new version of the module and simply runs the build step.
+* `test`: uses nyc to run the test suite, compiling ES2015 code on the fly.
+* `pretest`: I'm a fan of the [standard](https://github.com/feross/standard) code-style tool, and generally run this as a `pretest` hook.
+* `main`/`bin`: note that `main` and `bin` both reference the compiled code in the `lib/` folder.
```

Comment:
[cap] I would capitalize the first word of each item.

[clarify] The description for `build` is a little wordy. How about this instead?

> Uses Babel to transpile the ES2015 source into ES5 code, which is output in `lib/`.

[nit] Should the reference to "build step" in the `prepublish` description be "`build` step" instead?

[clarify] Should words "compiling" (`test` description) and "compiled" (`main`/`bin` description) be "transpiling" and "transpiled" instead?

[grammar] The `pretest` description would have two complete sentences if you just added the word "I" before the word "generally".